### ISSUE TITLE
Failure to present Create Ticket view

### ISSUE THREAD

**Issue by damonskinner** [zendesk/zendesk_sdk_ios#217]
For months, we had used Zendesk as part of our approach to customer service in an iOS app.  However, recently it has started to break and cause crashes.

Originally, the version of Zendesk we used was 1.5.4.1.  It worked perfectly fine when creating a customer service ticket view and allowing the user to enter feedback/comments, then submit them.  Then it started crashing 100% of the time upon instantiation of the view using ZDKRequests.showRequestCreationWithNavController(self.navigationController), due to this error:

*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'NSConcreteAttributedString initWithString:: nil value'
*** First throw call stack:
(
0   CoreFoundation                      0x0000000110709d85 __exceptionPreprocess + 165
1   libobjc.A.dylib                     0x0000000113c0cdeb objc_exception_throw + 48
2   CoreFoundation                      0x0000000110709cbd +[NSException raise:format:] + 205
3   Foundation                          0x0000000112f1818b -[NSConcreteAttributedString initWithString:] + 140
4   Foundation                          0x0000000112f18051 -[NSConcreteAttributedString initWithString:attributes:] + 27
5   Beerhouse                           0x000000010f5394d6 -[ZDKCreateRequestView setupViewAppearance] + 1092
6   Beerhouse                           0x000000010f538d4c -[ZDKCreateRequestView initWithFrame:] + 118
7   Beerhouse                           0x000000010f564f5e -[ZDKCreateRequestViewController initWithSuccess:andError:] + 223
8   Beerhouse                           0x000000010f557a32 +[ZDKRequests showRequestCreationWithNavController:withSuccess:andError:] + 108
9   Beerhouse                           0x000000010f2d47e7 _TFFC9Beerhouse18HelpViewController13showContactUsFT_T_U_FT_T_ + 4135
10  Beerhouse                           0x000000010f25066f
_TFFC9Beerhouse12DataMediator21getAccountInformationFTSS15completionBlockFT_T__T_U_FTSbGSqVS_14ResponseObject__T_ + 4943
11  Beerhouse                           0x000000010f323914
_TFFC9Beerhouse17ShopifyAPIService22getCustomerInformationFTSS15completionBlockFT7successSb14responseObjectGSqVS_14ResponseObject__T__T_U_FGV9Alamofire8ResponsePs9AnyObject_CSo7NSError_T_ + 1556
12  Alamofire                           0x000000010feda1c7
_TTRXFo_oGV9Alamofire8ResponsePs9AnyObject_CSo7NSError__dT__XFo_iGS0_PS1__S2___dT__ + 151
13  Alamofire                           0x000000010fed8acd
_TFFFC9Alamofire7Request8responseuRxS_22ResponseSerializerTyperFT5queueGSqPSo17OS_dispatch_queue__18responseSerializerx17completionHandlerFGVS_8Responsewx16SerializedObjectwx11ErrorObject_T__DS0_U_FT_T_U_FT_T_ + 797
14  Alamofire                           0x000000010fed7314
_TPA__TFFFC9Alamofire7Request8responseuRxS_22ResponseSerializerTyperFT5queueGSqPSo17OS_dispatch_queue__18responseSerializerx17completionHandlerFGVS_8Responsewx16SerializedObjectwx11ErrorObject_T__DS0_U_FT_T_U_FT_T_ + 164
15  Alamofire                           0x000000010fe98807 _TTRXFo__dT__XFdCb__dT__ + 39
16  libdispatch.dylib                   0x000000011469dd9d _dispatch_call_block_and_release + 12
17  libdispatch.dylib                   0x00000001146be3eb _dispatch_client_callout + 8
18  libdispatch.dylib                   0x00000001146a61ef _dispatch_main_queue_callback_4CF + 1738
19  CoreFoundation                      0x00000001106630f9
__CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
20  CoreFoundation                      0x0000000110624b99 __CFRunLoopRun + 2073
21  CoreFoundation                      0x00000001106240f8 CFRunLoopRunSpecific + 488
22  GraphicsServices                    0x00000001153e9ad2 GSEventRunModal + 161
23  UIKit                               0x0000000111933f09 UIApplicationMain + 171
24  Beerhouse                           0x000000010f38d9c2 main + 114
25  libdyld.dylib                       0x00000001146f292d start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException

It looks like somewhere, 'attributes' is nil when trying to create the string.  Not sure what's the cause within the ZendeskSDK or my implementation.  

So I tried updating to the newest version (1.7.1.1).  Now, since 
ZDKRequests.showRequestCreationWithNavController(self.navigationController) is deprecated, we're using 
ZDKRequests.presentRequestCreationWithViewController(self.navigationController), but it ends in a black screen with bugged navigation buttons and title.  Please see the attached for an example. 

![simulator screen shot aug 26 2016 3 27 44 pm](https://cloud.githubusercontent.com/assets/10213943/18018097/10fd6de2-6ba3-11e6-96be-db75cb75d9ed.png)

Furthermore, when the button that would normally be titled 'Cancel' is tapped, the app crashes due to this error: 

`-[ZDKCreateRequestViewController dismiss]: unrecognized selector sent to instance 0x7fa8287b1940`

What's going on?

**Comment by RonanMcH** [comment]
Hi @damonskinner , 
Can I ask you how you have integrated the SDK (through cocoapods or manually)? It appears that the `ZendeskSDKStrings.bundle` has not been correctly added or linked to your project. This is why you see all the translation keys. This is the reason for the original crash on 1.5.4.1 also but the crash doesn't exist anymore on 1.7.2.1. 

The unrecognized selector crash, I believe is happening because you are passing a `UINavigationController` when the API is looking for a `UIVIewController`. 
Please try:
```
ZDKRequests.presentRequestCreationWithViewController(self)
```
instead of:
```
ZDKRequests.presentRequestCreationWithViewController(self.navigationController)
```

Please let us know if this fixes this crash.

Thanks,
Ronan.

**Comment by damonskinner** [comment]
HI Ronan,  
It's integrated through cocoapods with bundler.  

As far as the ViewContoller issues, yes, I first tried the normal UIViewController, but if memory serves, it came up with the same result (blank screen, wrong labels, crash upon tapping cancel).  Apologies, I can't say for certain that that was the case, as while trying to figure it out, I was trying out various Zendesk versions and project builds.  However, I believe that I found somewhere in your documentation reference to using a UINavigationController (it may just be leftover from previous zendesk versions).

Oddly enough, our testflight downloads with the newest project build work fine (using Zendesk v. 1.5.4.1 and the deprecated 'ZDKRequests.showRequestCreationWithNavController(self.navigationController)') when the same build when running on simulator did not.  Perhaps our current problem is just a simulator issue?  If we run into this again, I can post a new issue or update the current issue.
